File: informatics.mccme.ru/Python/Chapter 2/O/O.py
n = int(input())
n = n % (60 * 60 * 24)
res = str(n // (60 * 60))
res = res + ':'
n = n % (60 * 60)
m = n // 60
n = n % 60
res = res + str(m // 10)
res = res + str(m % 10)
res = res + ':'
res = res + str(n // 10)
res = res + str(n % 10)
print(res)
 

File: informatics.mccme.ru/Math/Arythmetic/O/O.cpp
#include <iostream>
#include <cstdio>

using namespace std;

int res = 0;

void gcd(int n, int m) {
	if (!n || !m) {
		return;
	}

	if (n < m)
		swap(n, m);

	res += n / m;
	n %= m;
	gcd(m, n);
}

int main() {
	freopen("input.txt", "r", stdin);
	freopen("output.txt", "w", stdout);

	int n, m;

	cin >> n >> m;

	gcd(n, m);

	cout << res;

	return 0;
}
